transport/src/main/java/io/netty/channel/IoHandler.java

```diff
@@ -29,14 +29,20 @@
      *
      * @return the number of {@link Channel} for which I/O was handled.
      */
-    int run(IoExecutionContext runner);
+    int run(IoExecutionContext context);
 
     /**
      * Wakeup the {@link IoHandler}, which means if any operation blocks it should be unblocked and
      * return as soon as possible.
      */
     void wakeup(boolean inEventLoop);
 
+    /**
+     * Close all registered {@code Channel}s as preparation to destroy this {@link IoHandler} soon.
+     * This method may be called multiple times.
+     */
+    void closeRegistered();
```

Comment:
Just a suggestion, feel free to ignore. The name `closeRegistered` is not very clear IMHO, and this method seems to be called in preparation of a shutdown, and followed by a call to `destroy`. I would consider renaming these two methods to `prepareToXXXX` and `XXXX`  where `XXXX` is either `destroy`, or `close` or `release`, etc.

Reply:
@stef1927 good idea... let me call it `prepareToDestroy`